File: app/frontend/src/types/chat/response.ts
import type { Message } from "@/types/message";

export interface ChatCreated {
  conversation_id: string;
}

export interface ChatRead {
  course_name: string;
  messages: Message[];
}

export interface ChatListItem {
  conversation_id: string;
  course_name: string;
  updated_at: string;
}

File: app/backend/schemas/chat/response.py
from datetime import datetime
from pydantic import BaseModel, Field
from app.backend.schemas.message.response import MessageRead

class ChatCreated(BaseModel):
    conversation_id: str

class ChatRead(BaseModel):
    course_name: str
    messages: list[MessageRead] = Field(default_factory=list)

class ChatListItem(BaseModel):
    conversation_id: str
    course_name: str
    updated_at: datetime
